api/filters/labels/labels.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package labels
+
+import (
+	"sigs.k8s.io/kustomize/api/filters/fsslice"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+type labelMap map[string]string
+
+// Filter sets labels.
+type Filter struct {
+	// Labels is the set of labels to apply to the inputs
+	Labels labelMap `yaml:"labels,omitempty"`
+
+	// FsSlice identifies the label fields.
+	FsSlice types.FsSlice
+}
+
+var _ kio.Filter = Filter{}
+
+func (f Filter) Filter(nodes []*yaml.RNode) ([]*yaml.RNode, error) {
+	for i := range nodes {
+		if err := f.run(nodes[i]); err != nil {
+			return nil, err
+		}
+	}
+	return nodes, nil
+}
+
+// run applies this to a single node.
+func (f Filter) run(node *yaml.RNode) error {
+	for key, value := range f.Labels {
```

Comment:
WDYT of sorting the keys first, then iterating over the keys.  This will ensure the output is deterministic and multiple hydrations will always result in the same output.

Reply:
done-ish.  not sure if you had something fancier in mind with the anon func